test/ng/compileSpec.js

```diff
@@ -12111,6 +12111,614 @@ describe('$compile', function() {
+    }));
+
+    it('should work indipendently of (ng-)attributes with the same name', inject(function($rootScope, $compile) {
+      element = $compile('<span ng-prop-asdf="asdf" ng-attr-asdf="foo" />')($rootScope);
+      $rootScope.asdf = 123;
+      $rootScope.$digest();
+      expect(element.prop('asdf')).toBe(123);
+      expect(element.attr('asdf')).toBe('foo');
+    }));
+
+    it('should disallow property binding on onclick', inject(function($compile, $rootScope) {
+      // All event prop bindings are disallowed.
+      $rootScope.onClickJs = function() {};
+      expect(function() {
+          $compile('<button ng-prop-onclick="onClickJs"></script>');
+        }).toThrowMinErr(
+          '$compile', 'nodomevents', 'Property bindings for HTML DOM event properties are disallowed.  ' +
+          'Please use the ng- versions (such as ng-click or ng-on-click instead of ng-prop-onclick) instead.');
+      expect(function() {
+          $compile('<button ng-prop-ONCLICK="onClickJs"></script>');
+        }).toThrowMinErr(
+          '$compile', 'nodomevents', 'Property bindings for HTML DOM event properties are disallowed.  ' +
+          'Please use the ng- versions (such as ng-click or ng-on-click instead of ng-prop-onclick) instead.');
+    }));
+
+    it('should process property binding in pre-linking phase at priority 100', function() {
+      module(function() {
+        directive('propLog', function(log) {
+          return {
+            compile: function($element, $attrs) {
+              log('compile=' + $element.prop('myName'));
+
+              return {
+                pre: function($scope, $element, $attrs) {
+                  log('preLinkP0=' + $element.prop('myName'));
+                  $rootScope.name = 'pre0';
+                },
+                post: function($scope, $element, $attrs) {
+                  log('postLink=' + $element.prop('myName'));
+                  $rootScope.name = 'post0';
+                }
+              };
+            }
+          };
+        });
+      });
+      module(function() {
+        directive('propLogHighPriority', function(log) {
+          return {
+            priority: 101,
+            compile: function() {
+              return {
+                pre: function($scope, $element, $attrs) {
+                  log('preLinkP101=' + $element.prop('myName'));
+                  $rootScope.name = 'pre101';
+                }
+              };
+            }
+          };
+        });
+      });
+      inject(function($rootScope, $compile, log) {
+        element = $compile('<div prop-log-high-priority prop-log ng-prop-my_name="name"></div>')($rootScope);
+        log('postCompile=' + element.prop('myName'));
+        $rootScope.name = 'angular';
+        $rootScope.$apply();
+        log('digest=' + element.prop('myName'));
+        expect(log).toEqual('compile=undefined; preLinkP101=undefined; preLinkP0=undefined; postLink=undefined; postCompile=undefined; digest=angular');
```

Comment:
Here's one I wasn't sure about...

With `ng-attr-*` the attribute is [evaluated on pre-link](https://github.com/angular/angular.js/blob/v1.7.2/src/ng/compile.js#L3397-L3400) so it is available to other link functions before the first `$watch` applies. I have not done that for `ng-prop-*`. Should we?

Reply:
I don't think that's very important for properties. Listening to attributes is / was quite important in AngularJS, but with props that's different.

Reply:
Just make sure it is documented, so that people don't assume it works the same as ngAttr.

Reply:
I think for `$attr` it is important because properties are put into `$attrs` which is part of the directive compile/link API. But there is no equivalent API for properties. Unless you consider `$element.prop/attr` the API...

I think I'd vote to make it the same as `ngAttr` just for consistency.  Unless someone can think of a case where assigning to a DOM property one extra time (init + `$watch`) would be an issue?